src/lib/parser_txdef.h

```diff
@@ -99,7 +101,21 @@ typedef struct {
 } oasis_tx_t;
 
 typedef struct {
-    oasis_tx_t oasis_tx;
+    publickey_t id;
+    // We are going to read dynamically like for stakingAmendCommissionSchedule
+    publickey_t node;
+    size_t nodes_length;
+    bool allow_entity_signed_nodes;
+} oasis_entity_t;
+
+typedef struct {
+
+    union {
+        oasis_tx_t oasis_tx;
+        oasis_entity_t oasis_entity;
+    };
+
+    bool is_tx;
```

Comment:
an `enum` would be better in case we need to extend this later on

Reply:
also it is a good idea that the first element ( 0 ) in the enum is `TYPE_UNKNOWN` and we later reject that during validation. When the parser is initialized, we set the `type` to `TYPE_UNKNOWN`